core/Plugin/Manager.php

```diff
@@ -1654,5 +1668,17 @@ public function loadPluginTranslations()
             $translator->addDirectory(self::getPluginDirectory($pluginName) . '/lang');
         }
     }
+
+    private function savePluginActivationTime($pluginName)
+    {
+        $optionName = self::LAST_PLUGIN_ACTIVATION_TIME_OPTION_PREFIX . $pluginName;
+        Option::set($optionName, time());
+    }
+
+    private function savePluginDeactivationTime($pluginName)
+    {
+        $optionName = self::LAST_PLUGIN_DEACTIVATION_TIME_OPTION_PREFIX . $pluginName;
+        Option::set($optionName, time());
+    }
```

Comment:
Should we clean up those entries whenever a plugin gets completely uninstalled?

Reply:
I think it might still be useful to know when the plugin was last there... might be more useful to know when it was installed/uninstalled. What do you think @tsteur?

Reply:
No preference here. Might be useful though to keep them and I can't think of some downside when keeping these entries.